testbed/core/utils/logging_filters.py

```diff
@@ -8,70 +8,84 @@
 """
 import logging
 import os
-import threading
+import contextvars
 
 
-# Thread-local storage for trace context
-_trace_local = threading.local()
+_trace_context_var = contextvars.ContextVar("trace_context", default=None)
```

Comment:
I think we should move the contextvars and management of them to Middleware.  The reason is that the middleware can be the utility that other things depend on.  If we needed the request method, request path or trace_id anywhere else that we didn't have it, a ContextMiddleware could be called by other code as well as LoggingFilter.

It also just seems a little more robust for the logging filter to get answers from the middleware, rather than for the middleware to import the logging filter to push data to it.  So this would reverse the import relationship.